### ISSUE TITLE
log_uniform min max unintuitive

### ISSUE THREAD

**Issue by dreamflasher** [wandb/client#507]
wandb, version 0.8.9
Python 3.7.4
Linux

### Description

Currently ```Log uniform. Number between exp(min) and exp(max) so that the logarithm of the return value is uniformly distributed.```
While it should rather be distributed between min and max (keeping the meaning of the terms min and max) – my expectation would have been, if I want my learning rate between 1e-8 and 1, I can just tell it to be between 1e-8 and 1, but I actually have to convert it by taking the log of my desired values, passing min: -8 and max: 0, so that I get the scaling between 1e-8 and 1. 

Solution: In the parsing of the function take the log of the min and max params.

**Comment by Danmou** [comment]
Note that you need to take the natural logarithm, so 1e-8 actually becomes -18.42. This is highly unintuitive!

**Comment by mboratko** [comment]
In addition, it changes the interpreted value of the parameter simply by changing the distribution, which is very strange. For example, if I have a script expecting `learning_rate` then changing a sweep script with
```
learning-rate:
  min: 1e-5
  max: 1e-1
```
to
```
learning-rate:
  distribution: log_uniform
  min: 1e-5
  max: 1e-1
```
would require me to change my training program.

**Comment by kengz** [comment]
One suggestion/feature request: for all the distributions with (natural) log, can we please add the log10 version? Currently they would be:
- `log10_uniform`
- `log10_normal`
- `q_log10_uniform`
- `q_log10_normal`

For reference, I dug up the [source code](https://github.com/wandb/client-ng/blob/4dbce5afe41624abc3a187b2d005cba5678ea672/wandb/sweeps/params.py#L181) where the log distributions are implemented.

**Comment by dreamflasher** [comment]
Push

**Comment by kengz** [comment]
Created a new PR above to add `log10_uniform` and `q_log10_uniform` since they are quite straightforward, but left out the `normal` counterparts since they aren't trivial to implement.

**Comment by manuel-delverme** [comment]
+1 to this, i've we think in base 10, not e :D

**Comment by manuel-delverme** [comment]
Up, this is made worse by the fact the UI works in base 10

**Comment by billray0259** [comment]
Has any progress been made on this issue?

**Comment by dasturge** [comment]
I just wanted to say that a better solution might be to allow users to input a different (mutually exclusive) min/max pairing (just a proposal for the naming):
exp_min:  0.000000001
exp_max: 0.0001

that way it would still be the same distribution (log_uniform, which it is) but one could declare the more intuitive min and max values.

**Comment by FabianWesth** [comment]
push

**Comment by vanpelt** [comment]
Hey Guys, sorry for the lack of W&B activity on this thread.  We're hearing you loud and clear, the current implementation is confusing.  We're deciding internally whether to prioritize a quick fix of just adding new keys or wait for a larger redesign with a new config format that fixes these unintuitive defaults.

**Comment by manuel-delverme** [comment]
What about adding a quickfix while we wait for the new design, so that we won't have this problem for 3 more years?

**Comment by vanpelt** [comment]
@manuel-delverme ouch 😭 .  We hear you loud and clear.  We're working on a fix in our [sweep repo](https://github.com/wandb/sweeps/pull/60/files), hopefully it's ready in time for the next release next week.

**Comment by dannygoldstein** [comment]
Hey everyone, we just merged in a PR to fix this in our sweeps repo. We have to deploy it to production and update the docs now. I'll comment here when thats done with pointers to the new docs.

**Comment by dreamflasher** [comment]
Oh wow! I didn't believe it in ever happening, anymore!

**Comment by dannygoldstein** [comment]
https://www.youtube.com/watch?v=1k8craCGpgs

**Comment by dannygoldstein** [comment]
> Oh wow! I didn't believe it in ever happening, anymore!

@dreamflasher https://www.youtube.com/watch?v=1k8craCGpgs

**Comment by hu-po** [comment]
[Documentation has been updated](https://docs.wandb.ai/guides/sweeps/configuration#distribution)! Check out the three new distributions: `log_uniform_values`, `q_log_uniform_values`, and `inv_log_uniform_values`.

**Comment by ZeguanXiao** [comment]
> [Documentation has been updated](https://docs.wandb.ai/guides/sweeps/configuration#distribution)! Check out the three new distributions: `log_uniform_values`, `q_log_uniform_values`, and `inv_log_uniform_values`.

How can I use these distributions to achieve a log10_uniform distribution as the comment below requested?

> One suggestion/feature request: for all the distributions with (natural) log, can we please add the log10 version? This is actually useful for cases where we express quantities in notation like `1e-4` for learning rates, epsilons, other hyperparameters.
> 
> This means the addition of the following distribution:
> 
> * `log10_uniform`
> * `log10_normal`
> * `q_log10_uniform`
> * `q_log10_normal`
> 
> For reference, I dug up the [source code](https://github.com/wandb/client/blob/dae543aa655380e0f1c316e90e8db34b645236c3/wandb/sweeps/params.py#L181) where the log distributions are implemented.

**Comment by aalok-sathe** [comment]
```
Not found
Oops! This page doesn't exist. Try going back to the [home page](https://docs.wandb.ai/).

You can learn how to make a 404 page like this in [Custom 404 Pages](https://gohugo.io/templates/404/).
```
